File: laravel_backend/config/natan/natural_query.php
<?php

declare(strict_types=1);

return [
    'rate_limit' => [
        'max_attempts' => (int) env('NATURAL_QUERY_MAX_ATTEMPTS', 20),
        'decay_minutes' => (int) env('NATURAL_QUERY_DECAY_MINUTES', 5),
    ],
    'blacklist' => [
        // Parole o frasi vietate (minuscole per semplicità di matching)
        'drop database',
        'delete from',
        'truncate',
        'shutdown',
        'rm -rf',
    ],
];

File: python_ai_service/app/config/natural_query.py
"""Natural language query configuration and guardrails."""
import os
from typing import List

# Rate limit config (shared with Laravel)
MAX_ATTEMPTS = int(os.getenv("NATURAL_QUERY_MAX_ATTEMPTS", "20"))
DECAY_MINUTES = int(os.getenv("NATURAL_QUERY_DECAY_MINUTES", "5"))

# Blacklisted phrases (lowercase for easier matching)
BLACKLIST: List[str] = [
    "drop database",
    "delete from",
    "truncate",
    "shutdown",
    "rm -rf",
]

